File: src/main/java/com/parrit/transformers/PairingBoardTransformer.java
package com.parrit.transformers;

import com.parrit.DTOs.PairingBoardDTO;
import com.parrit.DTOs.PersonDTO;
import com.parrit.entities.PairingBoard;
import com.parrit.entities.Person;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

public class PairingBoardTransformer {

    public static PairingBoardDTO transform(PairingBoard pairingBoard) {
        List<PersonDTO> personDTOs = PersonTransformer.transform(pairingBoard.getPeople());
        return new PairingBoardDTO(pairingBoard.getId(), personDTOs, pairingBoard.getName());
    }

    public static List<PairingBoardDTO> transform(List<PairingBoard> pairingBoards) {
        if(pairingBoards == null || pairingBoards.isEmpty()) return Collections.emptyList();
        return pairingBoards.stream()
                .map(PairingBoardTransformer::transform)
                .collect(Collectors.toList());
    }

    public static PairingBoard reverse(PairingBoardDTO pairingBoardDTO) {
        List<Person> reversePeople = PersonTransformer.reverse(pairingBoardDTO.getPeople());
        return new PairingBoard(pairingBoardDTO.getName(), reversePeople, pairingBoardDTO.getId());
    }

    public static List<PairingBoard> reverse(List<PairingBoardDTO> pairingBoardDTOs) {
        if(pairingBoardDTOs == null || pairingBoardDTOs.isEmpty()) return Collections.emptyList();
        return pairingBoardDTOs.stream()
            .map(PairingBoardTransformer::reverse)
            .collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/parrit/transformers/PairingBoardTransformer.kt
package com.parrit.transformers

import com.parrit.DTOs.PairingBoardDTO
import com.parrit.entities.PairingBoard

object PairingBoardTransformer {

    fun transform(pairingBoard: PairingBoard): PairingBoardDTO {
        val pairingBoardDTO = PairingBoardDTO()
        pairingBoardDTO.id = pairingBoard.id
        pairingBoardDTO.exempt = pairingBoard.isExempt
        pairingBoardDTO.name = pairingBoard.name
        pairingBoardDTO.people = PersonTransformer.transform(pairingBoard.people)
        return pairingBoardDTO
    }

    fun transform(pairingBoards: List<PairingBoard>?): List<PairingBoardDTO> {
        if (pairingBoards == null || pairingBoards.isEmpty()) return emptyList()
        return pairingBoards.map { transform(it) }
    }

    fun reverse(pairingBoardDTO: PairingBoardDTO): PairingBoard {
        val pairingBoard = PairingBoard()
        pairingBoard.id = pairingBoardDTO.id
        pairingBoard.name = pairingBoardDTO.name
        pairingBoard.isExempt = pairingBoardDTO.exempt
        pairingBoard.people = PersonTransformer.reverse(pairingBoardDTO.people)
        return pairingBoard
    }

    fun reverse(pairingBoardDTOs: List<PairingBoardDTO>?): MutableList<PairingBoard> {
        when {
            pairingBoardDTOs == null || pairingBoardDTOs.isEmpty() -> return mutableListOf()
            else -> return pairingBoardDTOs.map { reverse(it) }.toMutableList()
        }
    }
}
